### PR TITLE
Add Nim filetype

### PR DESCRIPTION
**Pull request by trustable-code**
Add support for the Nim programming language using the Python lexer.
Nim website: https://nim-lang.org/

Additional notes:  
PR #193 is another PR for Nim support. That one has an own lexer. "Nimrod" is the old name of the language.

### CODE REVIEW

**Comment by elextr** [comment]
The filetypes file filename has the wrong form for a [custom filetype](https://www.geany.org/manual/current/index.html#custom-filetypes) I'm surprised it worked.

**Comment by trustable-code** [comment]
@elextr Thanks. Now it should be correct.

**Comment by elextr** [comment]
@b4n, a quick survey shows Nim doesn't make it near the top of any of the programming language rankings, its no typescript or coffeescript or other high profile JS languages  (as flawed these lists are, not making any is likely significant).

But whilst in the past I have also tended to suggest the wiki I have sort of changed my mind a bit, really should Geany (or other tools) be picking winners, especially when it is only a filetypes.XXX.conf file. Its unlikely to need much support effort at that level.

Certainly more heavily integrated languages with lexers and parsers that need more support should be considered carefully, but one config file seems low impact.

However as there are already too many programming languages, maybe we need another group in filetypes.extensions, but thats another issue (#2087 actually :).

**Comment by trustable-code** [comment]
@b4n Thanks. I have removed the unneeded things as you have requested. I kept the line "type=0" because Nim does not allow tabulators.

**Comment by qqtop** [comment]
Hope this can be merged any time soon now.

File: data/filedefs/filetypes.Nim.conf [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+type=0
+
+[build-menu]
+
+# %f will be replaced by the complete filename
+# %e will be replaced by the filename without extension
+# (use only one of it at one time)
+compiler=nim c "%f"
+run_cmd=%e
```
Comment (ntrel) [inline data/filedefs/filetypes.Nim.conf]: This should use `"%e"` in case there are spaces in the path.

**Comment by ntrel** [comment]
Thanks @trustable-code. If you have time to fix the `%e` issue, that would be great. Otherwise I could fix it later and merge. @b4n's comments all seem to be resolved now (Nim rejects tabs).

File: data/filedefs/filetypes.Nim.conf (hunk 1 above)
Comment (elextr) [reply data/filedefs/filetypes.Nim.conf]: Probably should be "./%e" to work on many Linux systems.

**Comment by trustable-code** [comment]
@ntrel The PR is now updated.

**Pull request merged**